pybigquery/sqlalchemy_bigquery.py

```diff
@@ -267,47 +268,61 @@ def __init__(
     def dbapi(cls):
         return dbapi
 
-    def create_connect_args(self, url):
-        location, dataset_id, arraysize, credentials_path, default_query_job_config = parse_url(url)
+    @staticmethod
+    def _add_default_dataset_to_job_config(job_config, project_id, dataset_id):
+        # If dataset_id is set, then we know the job_config isn't None
+        if dataset_id:
+            # If project_id is missing, use defualt project_id
+            if not project_id:
+                _, project_id = auth.default()
 
-        self.arraysize = self.arraysize or arraysize
-        self.location = location or self.location
-        self.credentials_path = credentials_path or self.credentials_path
-        self.dataset_id = dataset_id
+            job_config.default_dataset = f'{project_id}.{dataset_id}'
 
-        if self.credentials_path:
-            client = bigquery.Client.from_service_account_json(
-                self.credentials_path,
-                location=self.location,
-                default_query_job_config=default_query_job_config
-            )
-        elif self.credentials_info:
-            scopes = (
+
+    def _create_client_from_credentials(self, credentials, default_query_job_config):
+        scopes = (
                 'https://www.googleapis.com/auth/bigquery',
                 'https://www.googleapis.com/auth/cloud-platform',
                 'https://www.googleapis.com/auth/drive'
             )
-            credentials = service_account.Credentials.from_service_account_info(
-                self.credentials_info
-            )
-            credentials = credentials.with_scopes(scopes)
-            client = bigquery.Client(
-                project=self.credentials_info.get('project_id'),
+        credentials = credentials.with_scopes(scopes)
+
+        self._add_default_dataset_to_job_config(default_query_job_config,
+                                        credentials.project_id, self.dataset_id)
+
+        return bigquery.Client(
+                project=credentials.project_id,
```

Comment:
Note: there are some kinds of credentials that don't have a `project_id` property. e.g. if someone uses `gcloud application-default login`.

Edit: Looks like it's not relevant, as this is only called with service account credentials.